plugins/inputs/passenger/passenger.go

```diff
@@ -0,0 +1,221 @@
+package passenger
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/influxdb/telegraf/plugins/inputs"
+	"golang.org/x/net/html/charset"
+)
+
+type passenger struct {
+	Command string
+}
+
+type info struct {
+	Passenger_version  string `xml:"passenger_version"`
+	Process_count      int    `xml:"process_count"`
+	Capacity_used      int    `xml:"capacity_used"`
+	Get_wait_list_size int    `xml:"get_wait_list_size"`
+	Max                int    `xml:"max"`
+	Supergroups        struct {
+		Supergroup []struct {
+			Name               string `xml:"name"`
+			Get_wait_list_size int    `xml:"get_wait_list_size"`
+			Capacity_used      int    `xml:"capacity_used"`
+			Group              []struct {
+				Enabled_process_count   int `xml:"enabled_process_count"`
+				Disabling_process_count int `xml:"disabling_process_count"`
+				Disabled_process_count  int `xml:"disabled_process_count"`
+				Capacity_used           int `xml:"capacity_used"`
+				Get_wait_list_size      int `xml:"get_wait_list_size"`
+				Processes_being_spawned int `xml:"processes_being_spawned"`
+				Processes               struct {
+					Process []*process `xml:"process"`
+				} `xml:"processes"`
+			} `xml:"group"`
+		} `xml:"supergroup"`
+	} `xml:"supergroups"`
+}
+
+type process struct {
+	Pid                   int    `xml:"pid"`
+	Concurrency           int    `xml:"concurrency"`
+	Sessions              int    `xml:"sessions"`
+	Busyness              int    `xml:"busyness"`
+	Processed             int    `xml:"processed"`
+	Spawner_creation_time int64  `xml:"spawner_creation_time"`
+	Spawn_start_time      int64  `xml:"spawn_start_time"`
+	Spawn_end_time        int64  `xml:"spawn_end_time"`
+	Last_used             int64  `xml:"last_used"`
+	Uptime                string `xml:"uptime"`
+	Code_revision         string `xml:"code_revision"`
+	Life_status           string `xml:"life_status"`
+	Enabled               string `xml:"enabled"`
+	Has_metrics           bool   `xml:"has_metrics"`
+	Cpu                   int64  `xml:"cpu"`
+	Rss                   int64  `xml:"rss"`
+	Pss                   int64  `xml:"pss"`
+	Private_dirty         int64  `xml:"private_dirty"`
+	Swap                  int64  `xml:"swap"`
+	Real_memory           int64  `xml:"real_memory"`
+	Vmsize                int64  `xml:"vmsize"`
+	Process_group_id      string `xml:"process_group_id"`
+}
+
+func (p *process) getUptime() int64 {
```

Comment:
The uptime is report as string like `12d 3h 4m 5s`. This func convert it to second to store into InfluxBD